app/tests/integration/nimbus/pages/experimenter/base.py

```diff
@@ -1,15 +1,53 @@
 from nimbus.pages.base import Base
 from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
 
 
 class ExperimenterBase(Base):
     """Experimenter Base page."""
 
     _save_btn_locator = (By.CSS_SELECTOR, "#save-button")
     _save_continue_btn_locator = (By.CSS_SELECTOR, "#save-and-continue-button")
+    _sidebar_details_link = (By.CSS_SELECTOR, 'a[data-testid="nav-details"]')
+    _sidebar_edit_branches_link = (By.CSS_SELECTOR, 'a[data-testid="nav-edit-branches"]')
     NEXT_PAGE = None
+    PAGE_TITLE = ""
+
+    def wait_for_locator(self, locator, description):
+        self.wait.until(
+            EC.presence_of_all_elements_located(locator),
+            message=f"{self.PAGE_TITLE}: could not find {description}",
+        )
+
+    def wait_for_and_find_element(self, locator, description):
+        self.wait_for_locator(locator, description)
+        return self.find_element(*locator)
+
+    def wait_for_and_find_elements(self, locator, description):
+        self.wait_for_locator(locator, description)
+        return self.find_elements(*locator)
```

Comment:
Dupe method?

Reply:
There's find_element and find_elements (plural), if that's what looks dupey?

Reply:
Oh you're right yes ;)